pkg/provisioningv2/rke2/planner/probe.go

```diff
@@ -81,7 +81,34 @@ func isCalico(controlPlane *rkev1.RKEControlPlane, runtime string) bool {
 		cni == "calico+multus"
 }
 
-func (p *Planner) addProbes(nodePlan plan.NodePlan, controlPlane *rkev1.RKEControlPlane, machine *capi.Machine) (plan.NodePlan, error) {
+// renderSecureProbe takes the existing argument value and renders a secure probe using the argument values and an error
+// if one occurred.
+func renderSecureProbe(arg interface{}, rawProbe plan.Probe, runtime string, defaultSecurePort string, defaultCertDir string, defaultCert string) (plan.Probe, error) {
+	securePort := getArgValue(arg, SecurePortArgument, "=")
+	if securePort == "" {
+		// If the user set a custom --secure-port, set --secure-port to an empty string so we don't override
+		// their custom value
+		securePort = defaultSecurePort
+	}
+	TLSCert := getArgValue(arg, TLSCertFileArgument, "=")
+	if TLSCert == "" {
+		// If the --tls-cert-file Argument was not set in the config for this component, we can look to see if
+		// the --cert-dir was set. --tls-cert-file (if set) will take precedence over --tls-cert-file
```

Comment:
`--tls-cert-file (if set) will take precedence over --tls-cert-file` Did you mean over `--cert-dir`?

Reply:
I'll open a 2.6.4 PR to fix this, ugh

Reply:
it's nit, so we're good!